process/sync/baseForkDetector.go

```diff
@@ -611,3 +613,69 @@ func (bfd *baseForkDetector) computeGenesisTimeFromHeader(headerHandler data.Hea
 	genesisTime := int64(headerHandler.GetTimeStamp() - headerHandler.GetRound()*uint64(bfd.rounder.TimeDuration().Seconds()))
 	return genesisTime
 }
+
+func (bfd *baseForkDetector) addHeader(
+	header data.HeaderHandler,
+	headerHash []byte,
+	state process.BlockHeaderState,
+	selfNotarizedHeaders []data.HeaderHandler,
+	selfNotarizedHeadersHashes [][]byte,
+	doJobOnBHProcessed func(data.HeaderHandler, []byte, []data.HeaderHandler, [][]byte),
+) error {
+
+	if check.IfNil(header) {
+		return ErrNilHeader
+	}
+	if headerHash == nil {
+		return ErrNilHash
+	}
+
+	err := bfd.checkBlockBasicValidity(header, headerHash, state)
```

Comment:
This can still return errors related to the current node state.

Reply:
I think that there are not related to the current state as all the computation are just related to the final block which could not be reverted anyway, and the other is related to the chronology round, genesis time which should be the same for every node and not related to its sync state. Anyway I have removed from worker the possibility to be returned an error from this call

Reply:
Great. Will re-evaluate at a later time if we really need to return error.